src/vshaxe/Main.hx

```diff
@@ -106,6 +108,27 @@ class Main {
 		maybeStartServer();
 	}
 
+	function setLanguageConfiguration():Void {
+		// based on https://github.com/microsoft/vscode/blob/bb02817e2e549fd88710d0e0a0336b80648e90b5/extensions/typescript-language-features/src/features/languageConfiguration.ts#L15
+		var defaultWordPattern = "(-?\\d*\\.\\d\\w*)|([^\\`\\~\\!\\@\\#\\%\\^\\&\\*\\(\\)\\-\\=\\+\\[\\{\\]\\}\\\\\\|\\;\\:\\'\\\"\\,\\.\\<\\>\\/\\?\\s]+)";
+		var wordPattern = defaultWordPattern + "|(@:\\w*)"; // metadata
+		languages.setLanguageConfiguration("haxe", {
+			wordPattern: new RegExp(wordPattern),
+			indentationRules: {
+				decreaseIndentPattern: new RegExp("^((?!.*?\\/\\*).*\\*\\/)?\\s*[\\}\\]].*$"),
+				increaseIndentPattern: new RegExp("^((?!\\/\\/).)*(\\{[^}\"'`]*|\\([^)\"'`]*|\\[[^\\]\"'`]*)$"),
+				indentNextLinePattern: new RegExp("(^\\s*(for|while|do|if|else|try|catch)|function)\\b(?!.*[;{}]\\s*(\\/\\/.*|\\/[*].*[*]\\/\\s*)?$)")
+			},
+			onEnterRules: [
```

Comment:
These seem rather susceptible to whitespace:

![](https://i.imgur.com/VuCB9xb.gif)

![](https://i.imgur.com/u7FlvWi.gif)